src/cpp/src/text_callback_streamer.cpp

```diff
@@ -26,19 +26,19 @@ StreamingStatus TextCallbackStreamer::write(int64_t token) {
         return set_streaming_status(on_finalized_subword_callback(res.str()));
     }
 
+    constexpr char replacement[] = "\xef\xbf\xbd";  // MSVC with /utf-8 fails to compile � directly with newline in string literal error.
+    if (text.size() >= 3 && text.compare(text.size() - 3, 3, replacement) == 0) {
+        m_decoded_lengths[m_decoded_lengths.size() - 1] = -1;
+        // Don't print incomplete text
+        return set_streaming_status(on_finalized_subword_callback(res.str()));
```

Comment:
`IterableStreamer` needs a similar change. Please add `ChunkStreamer` to tests as well.

Reply:
1. Turned out `IterableStreamer` already handles this situation correctly.

2. We talked with @as-suvorov, looks like he will add such unit-tests together with implementation of `TextStreamer` with vectorized write.